plugins/UM3NetworkPrinting/src/Cloud/CloudOutputDeviceManager.py

```diff
@@ -41,31 +41,34 @@ def __init__(self):
         self._output_device_manager = application.getOutputDeviceManager()
 
         self._account = application.getCuraAPI().account
-        self._account.loginStateChanged.connect(self._getRemoteClusters)
+        self._account.loginStateChanged.connect(self._onLoginStateChanged)
         self._api = CloudApiClient(self._account, self._onApiError)
 
         # When switching machines we check if we have to activate a remote cluster.
         application.globalContainerStackChanged.connect(self._connectToActiveMachine)
 
         # create a timer to update the remote cluster list
         self._update_timer = QTimer(application)
-        self._update_timer.setInterval(self.CHECK_CLUSTER_INTERVAL * 1000)
+        self._update_timer.setInterval(int(self.CHECK_CLUSTER_INTERVAL * 1000))
         self._update_timer.setSingleShot(False)
         self._update_timer.timeout.connect(self._getRemoteClusters)
 
+        # Make sure the timer is started in case we missed the loginChanged signal
+        self._onLoginStateChanged()
+
+    #  Called when the uses logs in or out
+    def _onLoginStateChanged(self) -> None:
+        if self._account.isLoggedIn and not self._update_timer.isActive():
```

Comment:
if the user is logged in and the timer is active we shouldn't remove all clusters

Reply:
That should not happen... but you are right.